src/CommonAccessors.jl

```diff
@@ -50,38 +37,35 @@ getTimestamp(v::DataLevel1) = v.timestamp
 """
 $SIGNATURES
 
-Get the timestamp from a DFGFactor object.
-
-DEPRECATED -> use getTimestamp instead.
+Set the timestamp of a DFGFactor object.
+Dev note:
+Since it is not mutable it has to return a new variable with the updated timestamp.
+Use `updateVariable!` to update it in the factor graph.
+TODO either this or we should make the field/variable mutable
 """
-function timestamp(v::DataLevel1)
-    @warn "timestamp deprecated, use getTimestamp instead"
-    getTimestamp(v)
+function setTimestamp(v::DFGVariable, ts::DateTime)
+    return DFGVariable(v.label, ts, v.tags, v.ppeDict, v.solverDataDict, v.smallData, v.bigData, v._dfgNodeParams)
 end
 
-"""
-$SIGNATURES
-
-Set the timestamp of a DFGFactor object.
-"""
-setTimestamp!(v::DataLevel1, ts::DateTime) = v.timestamp = ts
+function setTimestamp(v::DFGVariableSummary, ts::DateTime)
+    return DFGVariableSummary(v.label, ts, v.tags, v.estimateDict, v.softtypename, v.bigData, v._internalId)
+end
```

Comment:
are you happy with this behavior for `setTimestamp`, we can make a `setTimestamp!` that updates automatically under the hood

Reply:
yes I think so, we just need to document this clearly so people who are keeping a reference to a DFGVSummary dont get a surprise that it was the wrong one for some reason (changing the timestamp in this case).  I'm assuming this is due to immutable type.  Again, we just need to document -- perhaps just in the docstring here?

Reply:
opened issue to remind